apollo-federation/src/sources/connect/validation/connect.rs

```
@@ -210,6 +215,7 @@ impl<'schema> Connect<'schema> {
             .selection
             .variables()
             .chain(self.http.variables())
+            .chain(self.errors.variables())
             .collect::<HashSet<_>>();
         if all_variables.contains(&Namespace::Batch) && all_variables.contains(&Namespace::This) {
```

This makes me think... what actually _happens_ when you set a custom error with batch? Wouldn't you end up with arrays in extensions by accident?

We should make sure we have an example of the right way to use this and are validating correctly.

---

Yea, this feels like a documentation problem for sure to highlight that you should have separate error config on `@connect` on `$batch` resolvers.

You wouldn't even end up with an array in extensions... you'd get the default extensions because it would silently fail to convert it to an object.